src/hyppo/dos.asm

```diff
@@ -263,7 +263,8 @@ trap_dos_setname:
 ;;         ========================
 
 tdsnfailure:
-        lda dos_error_code
+        ;; save the error code so a later trap_dos_geterrorcode will return it
+        sta dos_error_code
```

Comment:
I haven't looked, but this might be a common enough usage, that we should have an entry point save_dos_error_code_and_return_from_trap_with_failure or similar?